cookbooks/cdo-repository/recipes/default.rb

```diff
@@ -3,44 +3,35 @@
 # Recipe:: default
 #
 
-include_recipe "cdo-github-access"
-
-template "/home/#{node[:current_user]}/.gemrc" do
-  source 'gemrc.erb'
-  user node[:current_user]
-  group node[:current_user]
+# Sync repo via SSH if key is provided.
+include_recipe 'cdo-github-access'
+has_ssh_key = node['cdo-github-access'] && node['cdo-github-access']['id_rsa'] != ''
```

Comment:
If the key isn't set, the default `node['cdo-repository']['url']` attribute will be used as the git clone url (set to `'https://github.com/code-dot-org/code-dot-org.git'` by default in `attributes.rb`), making it possible to use this cookbook to clone the public github repository without requiring our private credentials within the Chef environment.

I could refactor this for a bit more readability (possibly collapsing cdo-github-access cookbook into cdo-repository, or at least moving the https and ssh attribute-defaults in the same place in the code) if I spend more time on it.